R/pkg/R/client.R

```diff
@@ -60,6 +60,15 @@ generateSparkSubmitArgs <- function(args, sparkHome, jars, sparkSubmitOpts, pack
   combinedArgs
 }
 
+determineLauncher <- function(sparkSubmitBin, combinedArgs, capture = FALSE) {
```

Comment:
the common function could be like:
```
  launchScript(script) {
    if (windowsOS) use shell()
    else (system2()
  }
```
This makes it more general, not binding to launching SparkSubmit